### PR TITLE
Nested return stmt

### PR DESCRIPTION
**Pull request by sava17**
GL

### CODE REVIEW

File: src/main/java/astvisitors/Typechecker.java [hunk 1]
```diff
@@ -37,14 +58,46 @@ public void visit(Func_defNode node) {
...
+
+
     }
 
-    public void visit(Return_stmtNode node) {
-        node.getReturnExpr().accept(this);
-        node.setType(node.getReturnExpr().getType());
+    public void visit(Print_lNode node) {
+        node.getExprNode().accept(this);
+    }
```
Comment (joandrsn) [inline src/main/java/astvisitors/Typechecker.java]: Not merged correctly

File: src/main/java/astvisitors/Typechecker.java [hunk 2]
```diff
@@ -26,6 +26,27 @@ public void visit(BlockNode node) {
             node.getReturnstmtNode().accept(this);
         }
 
+        /*
+        if (!elseStmtWithReturnExist && node.getReturnstmtNode() == null) {
+            System.err.println("Return is not guaranteed, since there are no else block with return or an return outside nested scopes");
+        }
+        */
```
Comment (joandrsn) [inline src/main/java/astvisitors/Typechecker.java]: Remove dead code

File: src/main/java/astvisitors/Typechecker.java [hunk 3]
```diff
@@ -26,6 +26,27 @@ public void visit(BlockNode node) {
             node.getReturnstmtNode().accept(this);
         }
 
+        /*
```
Comment (sentoa) [inline src/main/java/astvisitors/Typechecker.java]: Remove out-commented shit

File: src/main/java/astvisitors/Typechecker.java [hunk 4]
```diff
@@ -37,14 +58,46 @@ public void visit(Func_defNode node) {
...
+                    if(if_stmtNode.getElseBlock().getReturnstmtNode() != null) {
+                        elseStmtWithReturnExist = true;
+                    }
+                }
+            }
+        }
+
+        /* If there are no else stmt in the outer most scope of the func def block and no other return stmt, the func def is not guaranteed to reach an return stmt */
+        if (!elseStmtWithReturnExist && node.getBlockNode().getReturnstmtNode() == null) {
+            System.err.println("Return is not guaranteed, since there are no return or an else block in the outer scope with return");
```
Comment (joandrsn) [inline src/main/java/astvisitors/Typechecker.java]: Use error system

File: src/main/java/astvisitors/Typechecker.java [hunk 5]
```diff
@@ -37,14 +58,46 @@ public void visit(Func_defNode node) {
             checkType(node, node.getBlockNode().getReturnstmtNode());
         }
 
-        if (isReservedKeyword(node.getId()))
-            ErrorHandler.reservedKeyword(node.getId());
+        /* Gets the if statements of the outermost block, and checks if there is an else with an return stmt */
+        boolean elseStmtWithReturnExist = false;
+        StmtsNode stmtsOfOutermostBlock = node.getBlockNode().getStmtsNode();
+        for(int i = 0; i < stmtsOfOutermostBlock.getChildCount(); i++){
```
Comment (joandrsn) [inline src/main/java/astvisitors/Typechecker.java]: Formatting (Spaces after `for` etc.) missing

**Pull request merged**